File: two_pointers/valid_palindrome/valid_palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        res = ""
        for char in s:
            if char.isalnum():
                res+=char.lower()
        print(res)
        start = 0
        end = len(res)-1
        while start < end:
            if not res[start] == res[end]:
                return False
            start+=1
            end-=1
        return True

File: two_pointers/valid_palindrome/valid_palindrome.java
package two_pointers.valid_palindrome;

class Solution {
    public boolean isPalindrome(String s) {
        String res = "";
        for (int i = 0; i < s.length(); i++) {
            if (Character.isLetterOrDigit(s.charAt(i))) res+=s.charAt(i);
        }
        res = res.toLowerCase();

        int start = 0;
        int end = res.length() - 1;
        while (start < end) {
            if (res.charAt(start) != res.charAt(end)) {
                return false;
            }
            start++;
            end--;
        }
        return true;
    }
}